### PR TITLE
feat: Add new side navigation item type "section-group"

### PR DESCRIPTION
**Pull request by rubencarvalho**
### Description

We have a long standing request for an addition to the side navigation pattern to cater for deeper navigation options.

To solve this, we're introducing a new item type in the `Side Navigation` component:
`"section-group"` , which aggregates a set of items that are conceptually related to each other, and can be displayed under a single heading to provide further organization.
 We can nest sections, links, link groups and expandable link groups within a section group depending on the information architecture needs.

Examples below:

Organized with section headers (2)
Organized with section headers (1) (1)

Related links, issue #, if available: n/a

### How has this been tested?

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/side-navigation/internal.tsx [hunk 1]
```diff
@@ -166,9 +176,10 @@ function Link({ definition, expanded, activeHref, fireFollow }: LinkProps) {
 
 interface SectionProps extends BaseItemComponentProps {
   definition: SideNavigationProps.Section;
+  variant: 'section' | 'section-group' | 'link-group' | 'expandable-link-group' | 'root';
```
Comment (fa7ad) [inline src/side-navigation/internal.tsx:179]: Would an `enum` make sense here?

**Review by fa7ad — APPROVED** [review]
Looks like you have some a11Y failures, is this unrelated? or is that a known regression?
Other than that, LGTM

File: src/side-navigation/internal.tsx (hunk 1 above)
Comment (just-boris) [reply src/side-navigation/internal.tsx:179]: We are not using typescript enums. Some reasons are highlighted there: https://stackoverflow.com/questions/52393730/typescript-string-literal-union-type-from-enum

**Comment by rubencarvalho** [comment]
> 

I’ve addressed the A11Y tests. It was indeed a incorrectly built demo page.

**Review by fa7ad — APPROVED** [review]

**Pull request merged**